### PR TITLE
Add create_soft_signal_r and creates_soft_signal_rw methods

### PR DESCRIPTION
**Pull request by jsouter**
Moved from dodal, have placed in core.signals as seemed logical to prevent circular imports, could be moved to sim_signal_backends.py

### CODE REVIEW

File: tests/core/test_signal.py [hunk 1]
```diff
@@ -119,3 +122,18 @@ async def test_set_and_wait_for_value():
     assert not st.done
     set_sim_put_proceeds(sim_signal, True)
     assert await time_taken_by(st) < 0.1
+
+
+async def test_create_soft_signal():
```
Comment (DominicOram) [inline tests/core/test_signal.py:127]: Could: I would suggest splitting these into separate tests by either:
* Using `pytest.mark.parametrize`
* Have a helper function that is the test and call it with each case

File: src/ophyd_async/core/signal.py
Comment (coretl) [inline src/ophyd_async/core/signal.py:1]: @DominicOram you mentioned you needed a way to get a value into a `SignalR`, should we change `create_soft_signal_r` to return `Tuple[SignalR[T], SimSignalBackend]` instead? Should we also do the same for `create_soft_signal_rw`?

File: src/ophyd_async/core/signal.py [hunk 2]
```diff
@@ -253,6 +253,18 @@ def set_sim_callback(signal: Signal[T], callback: ReadingValueCallback[T]) -> No
     return _sim_backends[signal].set_callback(callback)
 
 
+def create_soft_signal_rw(
```
Comment (coretl) [inline src/ophyd_async/core/signal.py:256]: Slight preference for:
```suggestion
def soft_signal_rw(
```

File: src/ophyd_async/core/signal.py
Comment (jsouter) [reply src/ophyd_async/core/signal.py:1]: Okay, have pushed with only soft_signal_r returning the tuple, can change it if we prefer to have the same interface.

Comment (coretl) [reply src/ophyd_async/core/signal.py:1]: > Do we need it for create_soft_signal_rw? Can't you just do a usual set in that case?

No real need for it here, only if we want to make the 2 functions have the same return type.

Comment (coretl) [reply src/ophyd_async/core/signal.py:1]: I guess we could call them `soft_signal_rw` and `soft_signal_r_and_backend` (or something similar) if we care about being explicit?

Comment (DominicOram) [reply src/ophyd_async/core/signal.py:1]: Sounds good to me, I have no strong opinions

**Comment by jsouter** [comment]
@gilesknap @coretl Add in option to pass an initial value to SimSignalBackend in the init method, and also supporting this in the soft_signal_* methods. self._initial_value gets set to the initial_value argument in SimSignalBackend's init, but the converter only converts it when connect is called, though I think this is fine if users are only accessing it through the public methods. soft_signal_r_and_backend still returns the backend, though this may not be required now.

**Comment by gilesknap** [comment]
> @gilesknap @coretl Add in option to pass an initial value to SimSignalBackend in the init method, and also supporting this in the soft_signal_* methods. self._initial_value gets set to the initial_value argument in SimSignalBackend's init, but the converter only converts it when connect is called, though I think this is fine if users are only accessing it through the public methods. soft_signal_r_and_backend still returns the backend, though this may not be required now.

Thanks James, I have rebased and used the new feature successfully.

**Comment by jsouter** [comment]
Following discussion with @coretl, SimSignalBackends no longer take a source/name at init, and all backends now provide source as a method, where the Signal classes pass a name to the backend. source is available from signals as a property.
This shouldn't affect the CA/PVA backend much, but for devices using SimSignalBackend, the source descriptor entry is now based on the device's name, so the source may no longer resemble proper PV names, and will be more like "soft://my-sim-motor" instead of "soft://MY-SIM-MOTOR:X_RBV" e.g.

**Review by evalott100 — APPROVED** [review]
Looks good!

File: src/ophyd_async/core/signal.py [hunk 3]
```diff
@@ -253,6 +253,34 @@ def set_sim_callback(signal: Signal[T], callback: ReadingValueCallback[T]) -> No
...
+def soft_signal_rw(
+    datatype: Optional[Type[T]],
+    name: str,
+    initial_value: Optional[T] = None,
+) -> SignalRW[T]:
+    """Creates a read-writable Signal with a SimSignalBackend"""
+    signal = SignalRW(
+        SimSignalBackend(datatype, initial_value)
+    )
+    signal.set_name(name)
```
Comment (coretl) [inline src/ophyd_async/core/signal.py]: Please can we make `name=None` be passed to these helpers, and if it isn't already there then add it as an optional arg to `Signal.__init__`

Comment (coretl) [reply src/ophyd_async/core/signal.py]: And also add this to `epics_signal_*` helpers

Comment (jsouter) [reply src/ophyd_async/core/signal.py]: Ah! Missed this comment somehow, might have to be a new PR.

**Pull request merged**